Crash when heartbeat.interval.ms set to non-zero value.

I'm using the latest release with with this config:
```
[jboss@debezium-7d4676df44-86w2p debezium]$ cat config/application.properties
debezium.sink.type=bigquerystream
debezium.sink.bigquerystream.dataset=qa_ew11
debezium.sink.bigquerystream.location=EU
debezium.sink.bigquerystream.project=qa-europe-west1-1-2c5eadfd
debezium.sink.bigquerystream.create-if-needed=true
debezium.sink.bigquerystream.upsert=true
debezium.sink.bigquerystream.upsert-keep-deletes=false
debezium.source.connector.class=io.debezium.connector.postgresql.PostgresConnector
debezium.source.offset.storage=org.apache.kafka.connect.storage.FileOffsetBackingStore
debezium.source.offset.storage.file.filename=/debezium/data/connection.offsets
debezium.source.offset.flush.interval.ms=5000
debezium.source.schema.history.internal=io.debezium.relational.history.MemorySchemaHistory
debezium.source.plugin.name=pgoutput
debezium.source.slot.name=debezium
debezium.source.publication.name=debezium
debezium.source.database.hostname=<db-address>
debezium.source.database.user=debezium
debezium.source.database.password=<passwoed>
debezium.source.database.dbname=<database>
debezium.source.topic.prefix=tutorial
debezium.source.table.include.list=public.test0,public.test1
```

Works as expected but I'm testing with infrequently updated tables which causes Postgres WAL files to pile up. I tried adding single paramater `debezium.source.heartbeat.interval.ms=5000`  which causes crash:
```
2025-05-01 12:23:12,294 INFO  [io.deb.con.pos.con.PostgresConnection] (pool-11-thread-1) Obtained valid replication slot ReplicationSlot [active=false, latestFlushedLsn=LSN{B7E/16004100}, catalogXmin=1112898]
2025-05-01 12:23:12,311 INFO  [io.deb.con.pos.PostgresConnectorTask] (pool-11-thread-1) Found previous offset PostgresOffsetContext [sourceInfoSchema=Schema{io.debezium.connector.postgresql.Source:STRUCT}, sourceInfo=source_info[server='tutorial'db='report_prod', lsn=LSN{B7E/16004100}, txId=1112899, messageType=INSERT, lastCommitLsn=LSN{B7E/16004100}, timestamp=2025-05-01T09:32:13.031680Z, snapshot=FALSE, schema=, table=], lastSnapshotRecord=false, lastCompletelyProcessedLsn=LSN{B7E/16004100}, lastCommitLsn=LSN{B7E/16004100}, streamingStoppingLsn=null, transactionContext=TransactionContext [currentTransactionId=null, perTableEventCount={}, totalEventCount=0], incrementalSnapshotContext=IncrementalSnapshotContext [windowOpened=false, chunkEndPosition=null, dataCollectionsToSnapshot=[], lastEventKeySent=null, maximumKey=null]]
2025-05-01 12:23:12,317 INFO  [io.deb.con.pos.PostgresConnectorTask] (pool-11-thread-1) user 'debezium' connected to database 'report_prod' on PostgreSQL 14.17 (Ubuntu 14.17-1.pgdg22.04+1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0, 64-bit with roles:
        role 'debezium' [superuser: false, replication: true, inherit: true, create role: false, create db: false, can log in: true]
2025-05-01 12:23:12,319 INFO  [io.deb.con.pos.con.PostgresConnection] (pool-11-thread-1) Obtained valid replication slot ReplicationSlot [active=false, latestFlushedLsn=LSN{B7E/16004100}, catalogXmin=1112898]
2025-05-01 12:23:12,353 INFO  [io.deb.uti.Threads] (pool-11-thread-1) Requested thread factory for component PostgresConnector, id = tutorial named = SignalProcessor
2025-05-01 12:23:12,372 INFO  [io.deb.uti.Threads] (pool-11-thread-1) Requested thread factory for component PostgresConnector, id = tutorial named = change-event-source-coordinator
2025-05-01 12:23:12,372 INFO  [io.deb.uti.Threads] (pool-11-thread-1) Requested thread factory for component PostgresConnector, id = tutorial named = blocking-snapshot
2025-05-01 12:23:12,380 INFO  [io.deb.uti.Threads] (pool-11-thread-1) Creating thread debezium-postgresconnector-tutorial-change-event-source-coordinator
2025-05-01 12:23:12,382 INFO  [io.deb.emb.asy.AsyncEmbeddedEngine] (pool-7-thread-1) All tasks have started successfully.
2025-05-01 12:23:12,382 INFO  [io.deb.emb.asy.AsyncEmbeddedEngine] (pool-7-thread-1) Engine state has changed from 'STARTING_TASKS' to 'POLLING_TASKS'
2025-05-01 12:23:12,383 INFO  [io.deb.emb.asy.AsyncEmbeddedEngine] (pool-7-thread-1) Using io.debezium.embedded.async.ParallelSmtAndConvertBatchProcessor processor
2025-05-01 12:23:12,494 INFO  [io.deb.pip.ChangeEventSourceCoordinator] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Metrics registered
2025-05-01 12:23:12,495 INFO  [io.deb.pip.ChangeEventSourceCoordinator] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Context created
2025-05-01 12:23:12,503 INFO  [io.deb.con.pos.PostgresSnapshotChangeEventSource] (debezium-postgresconnector-tutorial-change-event-source-coordinator) A previous offset indicating a completed snapshot has been found.
2025-05-01 12:23:12,503 INFO  [io.deb.con.pos.PostgresSnapshotChangeEventSource] (debezium-postgresconnector-tutorial-change-event-source-coordinator) According to the connector configuration no snapshot will be executed
2025-05-01 12:23:12,507 INFO  [io.deb.pip.ChangeEventSourceCoordinator] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Snapshot ended with SnapshotResult [status=SKIPPED, offset=PostgresOffsetContext [sourceInfoSchema=Schema{io.debezium.connector.postgresql.Source:STRUCT}, sourceInfo=source_info[server='tutorial'db='report_prod', lsn=LSN{B7E/16004100}, txId=1112899, messageType=INSERT, lastCommitLsn=LSN{B7E/16004100}, timestamp=2025-05-01T09:32:13.031680Z, snapshot=FALSE, schema=, table=], lastSnapshotRecord=false, lastCompletelyProcessedLsn=LSN{B7E/16004100}, lastCommitLsn=LSN{B7E/16004100}, streamingStoppingLsn=null, transactionContext=TransactionContext [currentTransactionId=null, perTableEventCount={}, totalEventCount=0], incrementalSnapshotContext=IncrementalSnapshotContext [windowOpened=false, chunkEndPosition=null, dataCollectionsToSnapshot=[], lastEventKeySent=null, maximumKey=null]]]
2025-05-01 12:23:12,510 INFO  [io.deb.pip.ChangeEventSourceCoordinator] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Connected metrics set to 'true'
2025-05-01 12:23:12,541 INFO  [io.deb.con.pos.PostgresSchema] (debezium-postgresconnector-tutorial-change-event-source-coordinator) REPLICA IDENTITY for 'public.test1' is 'DEFAULT'; UPDATE and DELETE events will contain previous values only for PK columns
2025-05-01 12:23:12,543 INFO  [io.deb.con.pos.PostgresSchema] (debezium-postgresconnector-tutorial-change-event-source-coordinator) REPLICA IDENTITY for 'public.test0' is 'DEFAULT'; UPDATE and DELETE events will contain previous values only for PK columns
2025-05-01 12:23:12,568 INFO  [io.deb.pip.sig.SignalProcessor] (debezium-postgresconnector-tutorial-change-event-source-coordinator) SignalProcessor started. Scheduling it every 5000ms
2025-05-01 12:23:12,569 INFO  [io.deb.uti.Threads] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Creating thread debezium-postgresconnector-tutorial-SignalProcessor
2025-05-01 12:23:12,570 INFO  [io.deb.pip.ChangeEventSourceCoordinator] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Starting streaming
2025-05-01 12:23:12,571 INFO  [io.deb.con.pos.PostgresStreamingChangeEventSource] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Retrieved latest position from stored offset 'LSN{B7E/16004100}'
2025-05-01 12:23:12,572 INFO  [io.deb.con.pos.con.WalPositionLocator] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Looking for WAL restart position for last commit LSN 'LSN{B7E/16004100}' and last change LSN 'LSN{B7E/16004100}'
2025-05-01 12:23:12,572 INFO  [io.deb.con.pos.con.PostgresReplicationConnection] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Initializing PgOutput logical decoder publication
2025-05-01 12:23:12,608 INFO  [io.deb.con.pos.con.PostgresConnection] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Obtained valid replication slot ReplicationSlot [active=false, latestFlushedLsn=LSN{B7E/16004100}, catalogXmin=1112898]
2025-05-01 12:23:12,610 INFO  [io.deb.jdb.JdbcConnection] (pool-15-thread-1) Connection gracefully closed
2025-05-01 12:23:12,630 INFO  [io.deb.uti.Threads] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Requested thread factory for component PostgresConnector, id = tutorial named = keep-alive
2025-05-01 12:23:12,631 INFO  [io.deb.uti.Threads] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Creating thread debezium-postgresconnector-tutorial-keep-alive
2025-05-01 12:23:12,632 INFO  [io.deb.con.pos.PostgresStreamingChangeEventSource] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Searching for WAL resume position
2025-05-01 12:23:12,635 INFO  [io.deb.con.pos.con.WalPositionLocator] (debezium-postgresconnector-tutorial-change-event-source-coordinator) First LSN 'LSN{B7E/16001D58}' received
2025-05-01 12:23:12,652 INFO  [io.deb.con.pos.con.WalPositionLocator] (debezium-postgresconnector-tutorial-change-event-source-coordinator) LSN after last stored change LSN 'LSN{B7E/160041F0}' received
2025-05-01 12:23:12,652 INFO  [io.deb.con.pos.PostgresStreamingChangeEventSource] (debezium-postgresconnector-tutorial-change-event-source-coordinator) WAL resume position 'LSN{B7E/160041F0}' discovered
2025-05-01 12:23:12,654 INFO  [io.deb.jdb.JdbcConnection] (pool-16-thread-1) Connection gracefully closed
2025-05-01 12:23:12,656 INFO  [io.deb.jdb.JdbcConnection] (pool-17-thread-1) Connection gracefully closed
2025-05-01 12:23:12,674 INFO  [io.deb.con.pos.con.PostgresReplicationConnection] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Initializing PgOutput logical decoder publication
2025-05-01 12:23:12,687 INFO  [io.deb.uti.Threads] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Requested thread factory for component PostgresConnector, id = tutorial named = keep-alive
2025-05-01 12:23:12,688 INFO  [io.deb.uti.Threads] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Creating thread debezium-postgresconnector-tutorial-keep-alive
2025-05-01 12:23:12,688 INFO  [io.deb.con.pos.PostgresStreamingChangeEventSource] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Processing messages
2025-05-01 12:23:12,689 INFO  [io.deb.con.pos.con.AbstractMessageDecoder] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Streaming requested from LSN LSN{B7E/16004100}, received LSN LSN{B7E/16001D58} identified as already processed
2025-05-01 12:23:12,715 INFO  [io.deb.con.pos.con.WalPositionLocator] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Message with LSN 'LSN{B7E/160041F0}' arrived, switching off the filtering
2025-05-01 12:23:17,900 INFO  [io.deb.con.com.BaseSourceTask] (pool-11-thread-1) 1 records sent during previous 00:00:05.89, last recorded offset of {server=tutorial} partition is {lsn_proc=12636162900464, messageType=COMMIT, lsn_commit=12636162900464, lsn=12636162900464, txId=1112899, ts_usec=1746091933031680}
2025-05-01 12:23:18,429 ERROR [io.deb.emb.asy.AsyncEmbeddedEngine] (pool-7-thread-1) Engine has failed with : java.util.concurrent.ExecutionException: com.google.cloud.bigquery.BigQueryException: Clustering is enabled but no fields were specified.
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
        at io.debezium.embedded.async.AsyncEmbeddedEngine.runTasksPolling(AsyncEmbeddedEngine.java:496)
        at io.debezium.embedded.async.AsyncEmbeddedEngine.run(AsyncEmbeddedEngine.java:220)
        at io.debezium.server.DebeziumServer.lambda$start$1(DebeziumServer.java:182)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: com.google.cloud.bigquery.BigQueryException: Clustering is enabled but no fields were specified.
        at com.google.cloud.bigquery.BigQueryRetryHelper.runWithRetries(BigQueryRetryHelper.java:59)
        at com.google.cloud.bigquery.BigQueryImpl.create(BigQueryImpl.java:305)
        at io.debezium.server.bigquery.StreamBigqueryChangeConsumer.createTable(StreamBigqueryChangeConsumer.java:246)
        at io.debezium.server.bigquery.StreamBigqueryChangeConsumer.getTable(StreamBigqueryChangeConsumer.java:261)
        at io.debezium.server.bigquery.StreamBigqueryChangeConsumer.uploadDestination(StreamBigqueryChangeConsumer.java:153)
        at io.debezium.server.bigquery.BaseChangeConsumer.handleBatch(BaseChangeConsumer.java:122)
        at io.debezium.embedded.async.ParallelSmtAndConvertBatchProcessor.processRecords(ParallelSmtAndConvertBatchProcessor.java:56)
        at io.debezium.embedded.async.AsyncEmbeddedEngine$PollRecords.doCall(AsyncEmbeddedEngine.java:1203)
        at io.debezium.embedded.async.AsyncEmbeddedEngine$PollRecords.doCall(AsyncEmbeddedEngine.java:1183)
        at io.debezium.embedded.async.RetryingCallable.call(RetryingCallable.java:47)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        ... 3 more
Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 400 Bad Request
POST https://bigquery.googleapis.com/bigquery/v2/projects/qa-europe-west1-1-2c5eadfd/datasets/qa_ew11/tables?prettyPrint=false
{
  "code": 400,
  "errors": [
    {
      "domain": "global",
      "message": "Clustering is enabled but no fields were specified.",
      "reason": "invalid"
    }
  ],
  "message": "Clustering is enabled but no fields were specified.",
  "status": "INVALID_ARGUMENT"
}
        at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:145)
        at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:118)
        at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:37)
        at com.google.api.client.googleapis.services.AbstractGoogleClientRequest$3.interceptResponse(AbstractGoogleClientRequest.java:479)
        at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1111)
        at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:565)
        at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:506)
        at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:616)
        at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.createSkipExceptionTranslation(HttpBigQueryRpc.java:239)
        at com.google.cloud.bigquery.BigQueryImpl$2.call(BigQueryImpl.java:309)
        at com.google.cloud.bigquery.BigQueryImpl$2.call(BigQueryImpl.java:306)
        at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:102)
        at com.google.cloud.bigquery.BigQueryRetryHelper.run(BigQueryRetryHelper.java:92)
        at com.google.cloud.bigquery.BigQueryRetryHelper.runWithRetries(BigQueryRetryHelper.java:50)
        ... 15 more

2025-05-01 12:23:18,431 INFO  [io.deb.emb.asy.AsyncEmbeddedEngine] (pool-7-thread-1) Engine state has changed from 'POLLING_TASKS' to 'STOPPING'
2025-05-01 12:23:18,433 INFO  [io.deb.con.com.BaseSourceTask] (pool-11-thread-1) Stopping down connector
2025-05-01 12:23:18,798 INFO  [io.deb.jdb.JdbcConnection] (pool-18-thread-1) Connection gracefully closed
2025-05-01 12:23:18,799 INFO  [io.deb.jdb.JdbcConnection] (pool-19-thread-1) Connection gracefully closed
2025-05-01 12:23:18,799 INFO  [io.deb.pip.ChangeEventSourceCoordinator] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Finished streaming
2025-05-01 12:23:18,800 INFO  [io.deb.pip.ChangeEventSourceCoordinator] (debezium-postgresconnector-tutorial-change-event-source-coordinator) Connected metrics set to 'false'
2025-05-01 12:23:18,801 INFO  [io.deb.pip.sig.SignalProcessor] (pool-11-thread-1) SignalProcessor stopped
2025-05-01 12:23:18,802 INFO  [io.deb.ser.DefaultServiceRegistry] (pool-11-thread-1) Debezium ServiceRegistry stopped.
2025-05-01 12:23:18,804 INFO  [io.deb.jdb.JdbcConnection] (pool-20-thread-1) Connection gracefully closed
2025-05-01 12:23:18,804 INFO  [io.deb.emb.asy.AsyncEmbeddedEngine] (pool-7-thread-1) Stopped task #1 out of 1 tasks (it took 371 ms to stop the task).
2025-05-01 12:23:18,806 INFO  [org.apa.kaf.con.sto.FileOffsetBackingStore] (pool-7-thread-1) Stopped FileOffsetBackingStore
2025-05-01 12:23:18,807 INFO  [io.deb.emb.asy.AsyncEmbeddedEngine] (pool-7-thread-1) Engine is stopped.
2025-05-01 12:23:18,807 INFO  [io.deb.emb.asy.AsyncEmbeddedEngine] (pool-7-thread-1) Engine state has changed from 'STOPPING' to 'STOPPED'
2025-05-01 12:23:18,807 ERROR [io.deb.ser.ConnectorLifecycle] (pool-7-thread-1) Connector completed: success = 'false', message = 'com.google.cloud.bigquery.BigQueryException: Clustering is enabled but no fields were specified.', error = 'com.google.cloud.bigquery.BigQueryException: Clustering is enabled but no fields were specified.': com.google.cloud.bigquery.BigQueryException: Clustering is enabled but no fields were specified.
        at com.google.cloud.bigquery.BigQueryRetryHelper.runWithRetries(BigQueryRetryHelper.java:59)
        at com.google.cloud.bigquery.BigQueryImpl.create(BigQueryImpl.java:305)
        at io.debezium.server.bigquery.StreamBigqueryChangeConsumer.createTable(StreamBigqueryChangeConsumer.java:246)
        at io.debezium.server.bigquery.StreamBigqueryChangeConsumer.getTable(StreamBigqueryChangeConsumer.java:261)
        at io.debezium.server.bigquery.StreamBigqueryChangeConsumer.uploadDestination(StreamBigqueryChangeConsumer.java:153)
        at io.debezium.server.bigquery.BaseChangeConsumer.handleBatch(BaseChangeConsumer.java:122)
        at io.debezium.embedded.async.ParallelSmtAndConvertBatchProcessor.processRecords(ParallelSmtAndConvertBatchProcessor.java:56)
        at io.debezium.embedded.async.AsyncEmbeddedEngine$PollRecords.doCall(AsyncEmbeddedEngine.java:1203)
        at io.debezium.embedded.async.AsyncEmbeddedEngine$PollRecords.doCall(AsyncEmbeddedEngine.java:1183)
        at io.debezium.embedded.async.RetryingCallable.call(RetryingCallable.java:47)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 400 Bad Request
POST https://bigquery.googleapis.com/bigquery/v2/projects/qa-europe-west1-1-2c5eadfd/datasets/qa_ew11/tables?prettyPrint=false
{
  "code": 400,
  "errors": [
    {
      "domain": "global",
      "message": "Clustering is enabled but no fields were specified.",
      "reason": "invalid"
    }
  ],
  "message": "Clustering is enabled but no fields were specified.",
  "status": "INVALID_ARGUMENT"
}
        at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:145)
        at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:118)
        at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:37)
        at com.google.api.client.googleapis.services.AbstractGoogleClientRequest$3.interceptResponse(AbstractGoogleClientRequest.java:479)
        at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1111)
        at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:565)
        at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:506)
        at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:616)
        at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.createSkipExceptionTranslation(HttpBigQueryRpc.java:239)
        at com.google.cloud.bigquery.BigQueryImpl$2.call(BigQueryImpl.java:309)
        at com.google.cloud.bigquery.BigQueryImpl$2.call(BigQueryImpl.java:306)
        at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:102)
        at com.google.cloud.bigquery.BigQueryRetryHelper.run(BigQueryRetryHelper.java:92)
        at com.google.cloud.bigquery.BigQueryRetryHelper.runWithRetries(BigQueryRetryHelper.java:50)
        ... 15 more

2025-05-01 12:23:18,822 INFO  [io.deb.ser.DebeziumServer] (main) Received request to stop the engine
2025-05-01 12:23:18,822 INFO  [io.deb.ser.DebeziumServer] (main) Cannot shut down engine now:
2025-05-01 12:23:18,850 INFO  [io.quarkus] (main) debezium-server-bigquery-dist stopped in 0.041s
[jboss@debezium-7d4676df44-86w2p debezium]$
```

I'd appreciate any help.

--- a/debezium-server-bigquery-sinks/src/main/java/io/debezium/server/bigquery/BaseRecordConverter.java
+++ b/debezium-server-bigquery-sinks/src/main/java/io/debezium/server/bigquery/BaseRecordConverter.java
@@ -173,6 +173,10 @@ public abstract class BaseRecordConverter implements RecordConverter {
 
   @Override
   public TableConstraints tableConstraints() {
+    if (this.keyFields().isEmpty()) {
+      return null;
+    }
+
     return
         TableConstraints.newBuilder()
             .setPrimaryKey(PrimaryKey.newBuilder().setColumns(this.keyFields()).build())
@@ -183,13 +187,13 @@ public abstract class BaseRecordConverter implements RecordConverter {
   public Clustering tableClustering(String clusteringField) {
     // special destinations like "heartbeat.topics"
     if (this.destination().startsWith("__debezium")) {
-      return Clustering.newBuilder().build();
+      return null;
     }
 
-    if (this.keySchema() == null) {
+    ArrayList<String> keyFields = this.keyFields();
+    if (keyFields.isEmpty()) {
       return Clustering.newBuilder().setFields(List.of(clusteringField)).build();
     } else {
-      ArrayList<String> keyFields = this.keyFields();
       // NOTE Limit clustering fields to 4. it's the limit of Bigquery
       List<String> clusteringFields = keyFields.stream().limit(3).collect(Collectors.toList());
       clusteringFields.add(clusteringField);

--- a/debezium-server-bigquery-sinks/src/main/java/io/debezium/server/bigquery/StreamBigqueryChangeConsumer.java
+++ b/debezium-server-bigquery-sinks/src/main/java/io/debezium/server/bigquery/StreamBigqueryChangeConsumer.java
@@ -250,7 +250,7 @@ public class StreamBigqueryChangeConsumer extends BaseChangeConsumer {
 
     TableInfo tableInfo = TableInfo.newBuilder(tableId, tableDefBuilder.build()).build();
     Table table = bqClient.create(tableInfo);
-    LOGGER.warn("Created table {} PK {}", table.getGeneratedId(), tableInfo.getTableConstraints());
+    LOGGER.warn("Created table {} PK {}", table.getGeneratedId(), table.getTableConstraints());
     return table;
   }
 

--- a/debezium-server-bigquery-sinks/src/main/java/io/debezium/server/bigquery/StreamBigqueryChangeConsumer.java
+++ b/debezium-server-bigquery-sinks/src/main/java/io/debezium/server/bigquery/StreamBigqueryChangeConsumer.java
@@ -250,7 +250,11 @@ public class StreamBigqueryChangeConsumer extends BaseChangeConsumer {
 
     TableInfo tableInfo = TableInfo.newBuilder(tableId, tableDefBuilder.build()).build();
     Table table = bqClient.create(tableInfo);
-    LOGGER.warn("Created table {} PK {}", table.getGeneratedId(), tableConstraints.getPrimaryKey());
+    if (tableConstraints != null) {
+      LOGGER.warn("Created table {} PK {}", table.getGeneratedId(), tableConstraints.getPrimaryKey());
+    } else {
+      LOGGER.warn("Created table {} PK []", table.getGeneratedId());
+    }
     // NOTE @TODO ideally we should wait here for streaming cache to update with the new table information 
     // but seems like there is no proper way to wait... 
     // Without wait consumer fails

--- a/debezium-server-bigquery-sinks/src/main/java/io/debezium/server/bigquery/StreamBigqueryChangeConsumer.java
+++ b/debezium-server-bigquery-sinks/src/main/java/io/debezium/server/bigquery/StreamBigqueryChangeConsumer.java
@@ -250,15 +250,7 @@ public class StreamBigqueryChangeConsumer extends BaseChangeConsumer {
 
     TableInfo tableInfo = TableInfo.newBuilder(tableId, tableDefBuilder.build()).build();
     Table table = bqClient.create(tableInfo);
-    if (tableConstraints != null) {
-      LOGGER.warn("Created table {} PK {}", table.getGeneratedId(), tableConstraints.getPrimaryKey());
-    } else {
-      LOGGER.warn("Created table {} PK []", table.getGeneratedId());
-    }
-    // NOTE @TODO ideally we should wait here for streaming cache to update with the new table information 
-    // but seems like there is no proper way to wait... 
-    // Without wait consumer fails
-
+    LOGGER.warn("Created table {} PK {}", table.getGeneratedId(), tableInfo.getTableConstraints());
     return table;
   }
 

--- a/debezium-server-bigquery-sinks/src/main/java/io/debezium/server/bigquery/StreamBigqueryChangeConsumer.java
+++ b/debezium-server-bigquery-sinks/src/main/java/io/debezium/server/bigquery/StreamBigqueryChangeConsumer.java
@@ -235,11 +235,17 @@ public class StreamBigqueryChangeConsumer extends BaseChangeConsumer {
   private Table createTable(TableId tableId, Schema schema, Clustering clustering, TableConstraints tableConstraints) {
 
     StandardTableDefinition.Builder tableDefBuilder = StandardTableDefinition.newBuilder().setSchema(schema);
+
     if (!config.common().nestedAsJson()) {
-      tableDefBuilder
-          .setTimePartitioning(timePartitioning)
-          .setClustering(clustering)
-          .setTableConstraints(tableConstraints);
+      tableDefBuilder.setTimePartitioning(timePartitioning);
+    }
+
+    if (tableConstraints != null) {
+      tableDefBuilder.setTableConstraints(tableConstraints);
+    }
+
+    if (clustering != null && clustering.getFields() != null && !clustering.getFields().isEmpty()) {
+      tableDefBuilder.setClustering(clustering);
     }
 
     TableInfo tableInfo = TableInfo.newBuilder(tableId, tableDefBuilder.build()).build();

